src/store/modules/util/actions.ts

```diff
@@ -26,6 +27,118 @@ const actions: ActionTree<UtilState, RootState> = {
     } catch (err) {
       console.error("error", err);
     }
+  },
+  async getFacilities({ commit, state, dispatch }, payload) {
+    let resp;
+
+    try{
+      const filters = state.currentLocationFilterSelected;
+      if (filters.productStoreId !== 'All') payload.inputFields.productStoreId = filters.productStoreId;
+      if (filters.facilityTypeId !== 'All') payload.inputFields.facilityTypeId = filters.facilityTypeId;
+
+      resp = await UtilService.getFacilities(payload);
+
+      if(resp.status === 200 && resp.data.docs?.length && resp.data.docs?.length > 0 && !hasError(resp)) {
+        let facilities = resp.data.docs;
+        const total = resp.data.count;
+
+        let facilityIds: any = new Set();
+        facilities.forEach((facility: any) => {
```

Comment:
Why this is needed? We may be getting list with unique ids?

Reply:
Sir, I am passing queries for distinct facilities but getting multiple duplicate values, that's why I have used this.